fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java

```diff
@@ -598,31 +597,8 @@ private void castLiteralToTargetColumnsType(InsertStmt insertStatement) {
 
     private OptExprBuilder fillDefaultValue(LogicalPlan logicalPlan, ColumnRefFactory columnRefFactory,
                                             InsertStmt insertStatement, List<ColumnRefOperator> outputColumns) {
```

Comment:
when match column by name, use the `query output column names` to produce the `column list` in the order of query, `query output column names` and `column list` are same.
so revert `fillDefaultValue` to the old logic code.